workbase/src/renderer/components/SchemaDesign/store/actions.js

```diff
@@ -388,6 +389,20 @@ export default {
     commit('selectedNodes', [node.id]);
   },
 
+  async [DEFINE_RULE]({ state, dispatch }, payload) {
+    const graknTx = await dispatch(OPEN_GRAKN_TX);
+
+    // define rule
+    await state.schemaHandler.defineRule(payload);
+
+    await dispatch(COMMIT_TX, graknTx)
+      .catch((e) => {
+        graknTx.close();
+        logger.error(e.stack);
+        throw e;
+      });
```

Comment:
I see that you have this block of code (that commit tx and handles exceptions if any) in many places, if it becomes too verbose maybe you can just make a `async commitTx()` helper method that you can call and it wil contain the exception handling and everything